src/Groups/matrices/form_group.jl

```diff
@@ -0,0 +1,629 @@
+      sol = zero_matrix(F,n,n)
+      for i in 1:n
+         sol[i,i] = e(K[i,r])
+      end
+      pos = n+1
+      if isodd(q)
+         for i in 1:n, j in i+1:n
+            sol[i,j] = e(K[pos,r])+w*e(K[pos+N,r])       # sol[i,j] = a_ij + w*b_ij
+            sol[j,i] = e(K[pos,r])-w*e(K[pos+N,r])
+            pos +=1
+         end
+      else
+         for i in 1:n, j in i+1:n
+            sol[i,j] = e(K[pos,r])+w*e(K[pos+N,r])       # sol[i,j] = a_ij + w*b_ij
+            sol[j,i] = e(K[pos,r])+e(K[pos+N,r])+w*e(K[pos+N,r])
+            pos +=1
+         end
+      end
+      push!(L,sol)
+   end
+
+   return L
+end
+
+# TODO 2nd approach: using MeatAxe GAP functionalities
+# TODO: these are not exported at the moment
+
+"""
+    function invariant_bilinear_form(G::MatrixGroup)
+
+Return an invariant bilinear form for the group `G`. It works only if the module induced by the action of `G` is absolutely irreducible.
+!!! warning "Note:"
+    At the moment, the output is returned of type `mat_elem_type(G)`.
+"""
+function invariant_bilinear_form(G::MatrixGroup)
+   V = GAP.Globals.GModuleByMats(GAP.Globals.GeneratorsOfGroup(G.X), G.mat_iso.fr.codomain)
+   B = GAP.Globals.MTX.InvariantBilinearForm(V)
+   return G.mat_iso(B)   
+end
+
+"""
+    function invariant_sesquilinear_form(G::MatrixGroup)
+
+Return an invariant sesquilinear (non bilinear) form for the group `G`. It works only if the module induced by the action of `G` is absolutely irreducible.
+!!! warning "Note:"
+    At the moment, the output is returned of type `mat_elem_type(G)`.
+"""
+function invariant_sesquilinear_form(G::MatrixGroup)
+   V = GAP.Globals.GModuleByMats(GAP.Globals.GeneratorsOfGroup(G.X), G.mat_iso.fr.codomain)
+   B = GAP.Globals.MTX.InvariantSesquilinearForm(V)
+   return G.mat_iso(B)   
+end
+
+"""
+    function invariant_quadratic_form(G::MatrixGroup)
+
+Return an invariant bilinear form for the group `G`. It works only if the module induced by the action of `G` is absolutely irreducible.
+!!! warning "Note:"
+    At the moment, the output is returned of type `mat_elem_type(G)`.
+"""
+function invariant_quadratic_form(G::MatrixGroup)
+   if iseven(characteristic(base_ring(G)))
+      V = GAP.Globals.GModuleByMats(GAP.Globals.GeneratorsOfGroup(G.X), G.mat_iso.fr.codomain)
+      B = GAP.Globals.MTX.InvariantQuadraticForm(V)
+      return _upper_triangular_version(G.mat_iso(B))
+   else
+      m = invariant_bilinear_form(G)
+      for i in 1:degree(G), j in i+1:degree(G)
+         m[i,j]+=m[j,i]
+         m[j,i]=0
+      end
+      return m
+   end
+end
+
+
+# TODO 3rd approach: using GAP package "forms"
+"""
+    preserved_quadratic_forms(G::MatrixGroup)
+
+Uses random methods to find all of the quadratic forms preserved by `G` up to a scalar (i.e. such that `G` is a group of similarities for the forms). Since the procedure relies on a pseudo-random generator, the user may need to execute the operation more than once to find all invariant quadratic forms.
```

Comment:
Please do not use such long lines. Let's wrap them earlier; Julia manual recommends 92 chars as limit, so we could use e.g. that.

Please look at the other docstrings and also rewrap them if needed.